cppcache/test/CacheableStringTests.cpp

```diff
@@ -224,4 +224,28 @@ TEST_F(CacheableStringTests, TestFromDataNonAsciiHuge) {
   EXPECT_EQ(utf8, str->value());
 }
 
+class CacheableStringSizeTests : public ::testing::TestWithParam<std::size_t> {
+};
+
+INSTANTIATE_TEST_SUITE_P(CacheableStringTests, CacheableStringSizeTests,
+                         ::testing::Values(0UL, 1UL, 3UL, 7UL, 15UL, 23UL, 31UL,
+                                           47UL, 63UL, 1025UL, 4097UL));
+
+TEST_P(CacheableStringSizeTests, TestObjectSize) {
+  auto size = GetParam();
+  std::string str(size, '_');
+  auto cacheable = CacheableString::create(str);
+  EXPECT_TRUE(cacheable);
+
+  auto expected_size = sizeof(CacheableString);
```

Comment:
If we are literally copying the internal logic of the method is it really a valid test? Is there another way test the response?

Could we provide a customized allocator that we use for checking allocations? 🤷

Reply:
Actually, it's a good idea!! It should be feasible

Reply:
I think if you use the the custom allocator to allocate a `std::string` in the test itself to get the allocation size you could probably safely assume the same allocations internally. It's possible that an optimization might differ though and internally a different size is allocated. In that case the custom allocator bits have to flow through `CacheableString` that might get tricky.